workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ScreenViewFactory.kt

```diff
@@ -1,89 +1,279 @@
+    ): ScreenViewFactory<ScreenT> =
+      LayoutScreenViewFactory(ScreenT::class, layoutId, constructor)
+
+    /**
+     * Creates a [ScreenViewFactory] that inflates [layoutId] to "show" renderings of type
+     * [ScreenT], but never updates the created view. Handy for showing static displays,
+     * e.g. when prototyping.
+     */
+    @Suppress("unused")
+    public inline fun <reified ScreenT : Screen> forStaticLayoutResource(
+      @LayoutRes layoutId: Int
+    ): ScreenViewFactory<ScreenT> = forLayoutResource(layoutId) { ScreenViewRunner { _, _ -> } }
+
+    /**
+     * Creates a [ScreenViewFactory] that builds [View] instances entirely from code,
+     * using a [ScreenViewRunner] created by [constructor] to update it.
+     */
+    @WorkflowUiExperimentalApi
+    public inline fun <reified ScreenT : Screen> forBuiltView(
+      crossinline buildView: (
+        initialRendering: ScreenT,
+        initialEnvironment: ViewEnvironment,
+        context: Context,
+        container: ViewGroup?
+      ) -> ScreenViewHolder<ScreenT>,
+    ): ScreenViewFactory<ScreenT> {
+      return object : ScreenViewFactory<ScreenT> {
+        override val type = ScreenT::class
+
+        override fun buildView(
+          initialRendering: ScreenT,
+          initialEnvironment: ViewEnvironment,
+          context: Context,
+          container: ViewGroup?
+        ): ScreenViewHolder<ScreenT> =
+          buildView(initialRendering, initialEnvironment, context, container)
+      }
+    }
+  }
 }
 
 /**
- * It is usually more convenient to use [WorkflowViewStub] or [DecorativeScreenViewFactory]
- * than to call this method directly.
- *
- * Finds a [ScreenViewFactory] to create a [View] to display the receiving [Screen].
- * The caller is responsible for calling [View.start] on the new [View]. After that,
- * [View.showRendering] can be used to update it with new renderings that
- * are [compatible] with this [Screen]. [WorkflowViewStub] takes care of this chore itself.
+ * Use the [ScreenViewFactoryFinder] in [environment] to return the [ScreenViewFactory]
+ * bound to the type of the receiving [Screen].
  *
- * @param viewStarter An optional wrapper for the function invoked when [View.start]
- * is called, allowing for last second initialization of a newly built [View].
- * See [ViewStarter] for details.
+ * - It is more common to use [WorkflowViewStub.show] than to call this method directly
+ * - Call [ScreenViewFactory.start] to create and initialize a new [View]
+ * - If you don't particularly need to mess with the [ScreenViewFactory] before creating
+ *   a view, use [Screen.buildView] instead of this method.
+ */
+@WorkflowUiExperimentalApi
+public fun <ScreenT : Screen> ScreenT.toViewFactory(
+  environment: ViewEnvironment
+): ScreenViewFactory<ScreenT> {
+  return environment[ScreenViewFactoryFinder].getViewFactoryForRendering(environment, this)
+}
+
+/**
+ * It is more common to use [WorkflowViewStub.show] than to call this method directly.
```

Comment:
I really want to emphasize that `WorkflowViewStub` is the 99% case, though. I think this waters down that message.

```
 * It is more common to use [WorkflowViewStub.show] than to call this method directly.
 *
 * Creates a [ScreenViewHolder] wrapping a [View] able to display a stream
 * of [ScreenT] renderings, starting with [initialRendering].
 *
 * To add more initialization behavior (typically a call to [WorkflowLifecycleOwner.installOn]),
 * provide a [viewStarter].
```

Reply:
My concern was that "it is more common" is declarative rather than imperative for code that doesn't exist yet. I want to make it clear why it _should_ be more common. Though reading back I realize my redraft didn't do that either - ha.

Reply:
I think we used to say something like _It is very rare to call this method directly, instead use [WorkflowViewStub.show]_, WDYT?